.github/workflows/ci.yml

```
@@ -22,10 +22,21 @@ jobs:
       uses: actions/checkout@v2
       with:
         submodules: true
-    - name: Build ${{ matrix.distro }} Docker image with ${{ matrix.compiler }}
+    - name: Prepare base image in Docker
+      run: docker build . -f utils/docker/Dockerfile.${{ matrix.distro }}
+                          --build-arg DAOS_DEPS_BUILD=no
+    - name: Build dependencies in Docker
+      run: docker build . -f utils/docker/Dockerfile.${{ matrix.distro }}
+                          --build-arg DEPS_JOBS=10 --build-arg DAOS_BUILD=no
```

I wonder if something like `$(nproc)` is better than a hard-coded `10` here.
```suggestion
                          --build-arg DEPS_JOBS=$(nproc) --build-arg DAOS_BUILD=no
```

Of course, even $(nproc) is only a rough guess.  The mixture of IO-boundedness and CPU-boundedness of our builds could influence the optimal value for our build.  But ultimately, that value ought to be a function of `$(nproc)`.  Some experimentation on the exact function would be necessary.  But short of that, probably `$(nproc)` is a better _guess_ than a hard-coded 10?

---

how would proc get set?  I could have put it as a variable in this file but that wouldn't have improved readability in any real way.  10 was finger in the air, based on their being two cores available and wanting some level of parallelism.  I suspect the whole operation is I/O bound so overcommitting cores by 5x is a reasonable first guess.

---

> how would proc get set?

`$(nproc)` not `$proc`:

```
$ nproc
8
```

on my 8-core laptop.

---

I didn't know that one.

Github actions will give us two cores, and I agree that some tuning here would be useful.

https://docs.github.com/en/actions/reference/specifications-for-github-hosted-runners

---

> Github actions will give us two cores, and I agree that some tuning here would be useful.

And our own hosted runners might give us more or less, which is why a function of $(nproc) is probably better.

Additionally, $(nproc) would scale to end-users build systems also.